### PR TITLE
udiskie: fix build

### LINKED ISSUE #56771: udiskie doesn't build on nixos-unstable
## Issue description

Here is a log of `nix build udiskie -f channel:nixos-unstable `
[udiskie.log](https://github.com/NixOS/nixpkgs/files/2922856/udiskie.log)

CC @AndersonTorres 

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.12, NixOS, 19.09.git.e0f1c42 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(teto): `"nixos-18.09, nixos-unstable"`
 - nixpkgs: `/home/teto/nixpkgs`

```

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
It was broken by b4acd9772975d549f491d48debb679cf4ec78133.
Fixes #56771.

I'm not sure I know enough about GTK to determine which dependencies belong where. Feedback is welcome.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @teto

### CODE REVIEW

**Review by AndersonTorres — APPROVED** [review]
LGTM!

File: pkgs/applications/misc/udiskie/default.nix [hunk 1]
```diff
@@ -18,16 +15,27 @@ buildPythonApplication rec {
...
     asciidoc-full        # For building man page.
-    hicolor-icon-theme
+    gobject-introspection
     wrapGAppsHook
+  ];
+
+  buildInputs = [
+    hicolor-icon-theme
     librsvg              # required for loading svg icons (udiskie uses svg icons)
+    gobject-introspection
```
Comment (worldofpeace) [inline pkgs/applications/misc/udiskie/default.nix]:
```suggestion
```

**Comment by worldofpeace** [comment]
We have `python-dbusmock` so maybe enable the tests?

File: pkgs/applications/misc/udiskie/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/applications/misc/udiskie/default.nix]: That yields
```
$ udiskie -t &
Typelib for 'libnotify' is not available. Possible causes include:
	- libnotify is not installed
	- the typelib is provided by a separate package
	- libnotify was built with introspection disabled

Starting udiskie without notifications.
Typelib for 'Gtk 3.0' is not available. Possible causes include:
	- GTK3 is not installed
	- the typelib is provided by a separate package
	- GTK3 was built with introspection disabled
Starting udiskie without tray icon.
```
That's why I put it in both.

Comment (jtojnar) [reply pkgs/applications/misc/udiskie/default.nix]: Weird, I would expect setup hooks to work front `nativeBuildInputs` even with `strictDeps`.

**Comment by teto** [comment]
Binaries seem to work and it builds so fine by me.

**Comment by xeji** [comment]
@dotlambda It does not seem broken on 19.03: https://hydra.nixos.org/job/nixos/release-19.03/nixpkgs.udiskie.x86_64-linux/all - should we backport anyway?

**Comment by dotlambda** [comment]
> @dotlambda It does not seem broken on 19.03: https://hydra.nixos.org/job/nixos/release-19.03/nixpkgs.udiskie.x86_64-linux/all - should we backport anyway?

It's broken on `staging-19.03`. I'll do the backport.

**Comment by IvanMalison** [comment]
@dotlambda I'm not 100% sure about this, but can you explain why you moved libappindicator-gtk3 out of the propagated build inputs. My recollection is that the appindicator functionality does not work without it being there

**Comment by jtojnar** [comment]
`propagatedBuildInputs` are not of any use for applications (ignoring Python dependencies). Appindicator is actually used through `typelib` picked up by `gobject-introspection`.

**Comment by dotlambda** [comment]
As stated above,
> I'm not sure I know enough about GTK to determine which dependencies belong where.

I simply moved all non-Python dependencies out of `propagatedBuildInputs`.

**Pull request merged**